libmattermost.c

```diff
@@ -1196,6 +1217,40 @@ mm_get_users_by_ids(MattermostAccount *ma, GList *ids)
 	g_free(url);       
 }
 
+#define _MM_TOOLTIP_LINE_ADD(b,u,d,p,o) \
+{ \
+	if (o) { \
+		purple_notify_user_info_add_pair(u,d,o); \
+	} else { \
+		const gchar *v = purple_blist_node_get_string(PURPLE_BLIST_NODE(b),p); \
+		if (v && *v) { \
+			purple_notify_user_info_add_pair(u,d,v); \
+		} \
+	} \
+}
+
+static void 
+mm_tooltip_text(PurpleBuddy *buddy,PurpleNotifyUserInfo *user_info,gboolean full)
+{
+
+	const PurplePresence *presence = purple_buddy_get_presence(buddy);
+	MattermostAccount *ma = purple_connection_get_protocol_data(buddy->account->gc);

```

Comment:
This bit didn't work in purple-3, as the PurpleBuddy struct isn't opaque any more